components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -197,6 +198,27 @@ pub fn right_utf8(lhs: &Option<Bytes>, rhs: &Option<Int>) -> Result<Option<Bytes
     }
 }
 
+#[rpn_fn(capture = [args])]
+#[inline]
+pub fn upper(args: &[RpnStackNode<'_>], arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    match arg.as_ref() {
+        Some(bytes) => match args[0] {
+            RpnStackNode::Scalar { field_type, .. } => {
+                if field_type.as_accessor().is_binary_string_like() {
```

Comment:
@lonng 
> 
> We should not check the return field type of Upper expression, but should check the child return filed type. We cannot do this in the current implementation and we will support it in recent.

I fix it and make the unit tests pass. But the integration tests fail:
```
[2019-12-10T14:26:00.868Z] 2019/12/10 22:26:00 2019/12/10 22:25:45 Test fail: Outputs are not matching.

[2019-12-10T14:26:00.868Z] Test case: sql/randgen/7_str_2.sql

[2019-12-10T14:26:00.868Z] Statement: #1536 -  SELECT CONCAT( `col_decimal_unsigned` ) AS field1 FROM `table20_int_autoinc` WHERE UPPER( `col_varchar_64_key` ) /* QNO 1538 CON_ID 176 */ ;

[2019-12-10T14:26:00.868Z] NoPushDown Output: 

[2019-12-10T14:26:00.868Z] field1

[2019-12-10T14:26:00.868Z] 0

[2019-12-10T14:26:00.868Z] 0

[2019-12-10T14:26:00.868Z] 6641

[2019-12-10T14:26:00.868Z] 0

[2019-12-10T14:26:00.868Z] 1

[2019-12-10T14:26:00.868Z] NULL

[2019-12-10T14:26:00.868Z] 13939

[2019-12-10T14:26:00.868Z] 

[2019-12-10T14:26:00.868Z] 

[2019-12-10T14:26:00.868Z] PushDownWithoutVec Output: 

[2019-12-10T14:26:00.868Z] field1

[2019-12-10T14:26:00.868Z] 0

[2019-12-10T14:26:00.868Z] 0

[2019-12-10T14:26:00.868Z] 6641

[2019-12-10T14:26:00.868Z] 0

[2019-12-10T14:26:00.868Z] 1

[2019-12-10T14:26:00.868Z] NULL

[2019-12-10T14:26:00.868Z] 13939

[2019-12-10T14:26:00.868Z] 

[2019-12-10T14:26:00.868Z] 

[2019-12-10T14:26:00.868Z] PushDownWithVec Output: 

[2019-12-10T14:26:00.868Z] field1

[2019-12-10T14:26:00.868Z] 

[2019-12-10T14:26:00.868Z] 

[2019-12-10T14:26:00.868Z] 
```

On the other hand, I find the previous implementation makes integration tests pass :(